gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/errors/exceptions/exceptions.py

```diff
@@ -97,3 +97,10 @@ class GenAIPromptTemplateException(GenAIOrchestratorException):
 
     def __init__(self, info: ErrorInfo):
         super().__init__(ErrorCode.GEN_AI_PROMPT_TEMPLATE_ERROR, info)
+
+
+class CompressorUnknownException(GenAIOrchestratorException):
```

Comment:
About exceptions it's not really in this part but you should create an handler in `gen_ai_orchestrator/errors/handlers` that catches the FlashRankRerank exceptions but I'm not sure they have specific exception managed in their library